Source/MMWormhole.h

```diff
@@ -149,6 +167,52 @@ FOUNDATION_EXPORT const unsigned char MMWormholeVersionString[];
  */
 - (void)stopListeningForMessageWithIdentifier:(nullable NSString *)identifier;
 
-NS_ASSUME_NONNULL_END
+@end
+
+
+/**
+ This protocol defines the public interface for classes wishing to support the transiting of data
+ between two sides of the wormhole. Transiting is defined as passage between two points, and in this
+ case it involves both the reading and writing of messages as well as the deletion of message
+ contents.
+ */
+@protocol MMWormholeTransiting <NSObject>
+
+/**
+ This method is responsible for writing a given message object in a persisted format for a given
+ identifier. The method should return YES if the message was successfully saved. The message object 
+ may be nil, in which case YES should also be returned. Returning YES from this method results in a 
+ notification being fired which will trigger the corresponding listener block for the given 
+ identifier.
+ 
+ @param messageObject The message object to be passed.
+ This object may be nil. In this the method should return YES.
+ @param identifier The identifier for the message
+ @return YES indicating that a notification should be sent and NO otherwise
+ */
+- (BOOL)writeMessageObject:(nullable id)messageObject forIdentifier:(NSString *)identifier;
```

Comment:
Even though to chatted about this offline, adding this as a remember. The `id` should conform to `NSCoding`.